unlock-app/src/components/creator/lock/LockIconBar.js

```diff
@@ -23,28 +23,19 @@ export function LockIconBar({
   config,
   edit,
 }) {
-  // If the lockCreationTransaction does not exist, let's assume the lock is "pending", otherwise we
-  // would have its lockCreationTransaction.
-  if (!lockCreationTransaction) {
-    return <CreatorLockStatus lock={lock} status="Submitted" />
-  }
-
   // These 2 transactions, if not mined or confirmed will trigger the display of CreatorLockStatus
   // instead of the regular iconBar
   const blockingTransactions = [
     lockCreationTransaction,
     priceUpdateTransaction,
   ].filter(t => !!t)
 
-  // TODO: move that logic to mapStateToProps
+  // If there is any blocking transaction, we show the lock as either submitted or confirming
   for (let i = 0; i < blockingTransactions.length; i++) {
```

Comment:
This logic looks like it will consider at most one element from the list of blocking transactions? Does it need to be a loop then?